pkg/controller/storagecapabilities.go

```diff
@@ -53,13 +56,49 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"csi.trident.netapp.io/ontap-san": {{AccessMode: v1.ReadWriteOnce, VolumeMode: v1.PersistentVolumeBlock}},
 }
 
-// Get finds and returns a predefined StorageCapabilities for a given StorageClass
-func Get(sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
+// GetStorageCapabilities finds and returns a predefined StorageCapabilities for a given StorageClass
+func GetStorageCapabilities(cl client.Client, sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
 	provisionerKey := storageProvisionerKey(sc)
+	if provisionerKey == "kubernetes.io/no-provisioner" {
+		return capabilitiesForNoProvisioner(cl, sc)
+	}
 	capabilities, found := CapabilitiesByProvisionerKey[provisionerKey]
 	return capabilities, found
 }
 
+func capabilitiesForNoProvisioner(cl client.Client, sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
+	pvs := &v1.PersistentVolumeList{}
+	err := cl.List(context.TODO(), pvs)
+	if err != nil {
+		return []StorageCapabilities{}, false
+	}
+	capabilities := []StorageCapabilities{}
+	for _, pv := range pvs.Items {
+		if pv.Spec.StorageClassName == sc.Name {
+			for _, accessMode := range pv.Spec.AccessModes {
+				capabilities = append(capabilities, StorageCapabilities{
+					AccessMode: accessMode,
+					VolumeMode: resolveVolumeMode(pv.Spec.VolumeMode),
+				})
```

Comment:
Nice, we pair each available accessMode with the volumeMode. That is how it looks like for ceph as well:
```bash
  - accessModes:
    - ReadWriteMany
    volumeMode: Block
  - accessModes:
    - ReadWriteOnce
    volumeMode: Block
```